### ISSUE TITLE
bug: insert mode highligted cursors and higlighted background on current line

### ISSUE THREAD

**Issue by omar-zeineddine** [vscode-neovim/vscode-neovim#1352]
### Did you check docs and existing issues?

- [X] I have read all the vscode-neovim docs
- [X] I have searched the existing issues of vscode-neovim
- [X] I can reproduce the issue with an empty init.vim/init.lua and no other vscode extensions (when applicable)

### Neovim version (nvim -v)

0.9.1

### Operating system/version

linux

### Describe the bug

insert mode behaving like visual mode
new edited line displaying a background color, end cursor showing at the end of previous lines

![2](https://github.com/vscode-neovim/vscode-neovim/assets/83631525/d6b2d303-bc67-4915-9b51-cba94cb350a9)

### Steps To Reproduce

1. open vscode
2. insert mode on any file

### Expected Behavior

normal editing in insert mode

**Comment by theol0403** [comment]
Hmm, this is probably caused by a highlight group that we don't ignore anymore.

I'll make you a custom build in a bit to collect some logs.

**Comment by crash5** [comment]
For me the random vertical "highlighted boxes" as the start or end of the lines are caused by the `colorcolumn` neovim setting. When I set it to empty string the boxes are gone. I have no idea for the highlighted line. (Maybe some cursorline setting or plugin with the same functionality?)

**Comment by theol0403** [comment]
Yeah, don't turn on `colorcolumn`. @omar-zeineddine try to figure out which setting or plugin is causing this, although you did indicate you tried with empty init.vim.

I opted *not* to ignore those groups by default, to allow plugins to link to those styles properly. However, it does mean that your vim config has the ability to create more garbage.

**Comment by delinx** [comment]
same issue on my setup, worked fine for months up until few days ago

**Comment by omar-zeineddine** [comment]
Disabling the colorcolumn settings fixed this.

**Comment by theol0403** [comment]
I'll ask for some feedback about these changes. Previously, `CursorColumn` was ignored by vscode-neovim. However, this means any plugin that wants to link to `hi link CursorColumn` is broken. Additionally, you should never enable neovim settings that don't make sense in vscode-neovim, this is unsupported behavior.

**Pros of ignoring `CursorColumn`**
- People who have `cursorcolumn` set (even though there is no reason for them to) don't see a visual bug
**Cons of ignoring `CursorColumn`
- Plugins/config that link to `CursorColumn` don't work

Thumbs up if you are good with the new behavior, thumbs down if you want it to be ignored by default.

**Comment by theol0403** [comment]
Actually, I think a better fix would be to `set nocursorcolumn` by default.

**Comment by solson** [comment]
`set nocursorcolumn` does not fix it for me, but `set colorcolumn=` in my vscode-specific neovim config does.

**Comment by solson** [comment]
This visual bug was very janky and very confusing — red blocks were appearing and disappearing as I scrolled in an unpredictable manner, and I had to disable all and re-enable vscode extensions gradually to even figure out that vscode-neovim was the cause.

> Additionally, you should never enable neovim settings that don't make sense in vscode-neovim, this is unsupported behavior.

However, this is quite fair... I was less careful than I realized when migrating to vscode-neovim. Some of my config is loaded in both vscode and non-vscode neovim, and I clearly missed some things that should be moved from the shared to the non-vscode-only config.

On balance, I think it would indeed be best to `set colorcolumn=` by default, since the visual bug is so confusing. (It would be ideal to *explicitly warn* users when they set inappropriate config for vscode-neovim, but I imagine that would be a lot of work.)

Thanks all for investigating this!

**Comment by mschwartz** [comment]
Awesome that there is a workaround for this.

These random red blocks almost made me switch to terminal nvim until a new version of this plugin is available.  I'm glad I found this issue.

FWIW, I created an init.vscode.vim in my .config/nvim directory and use that for this plugin.  It's really stripped down and includes none of the plugins I normally install/use with nvim.  So it's basically a bunch of settings, including setting colorcolumn.  

Way easier to debug things by adding from my normal init.vim to the init.vscode.vim over time.
